frontend/src/metabase/visualizations/shared/settings/cartesian-chart.ts

```diff
@@ -0,0 +1,74 @@
+import _ from "underscore";
+import type { ComputedVisualizationSettings } from "metabase/visualizations/types";
+import type { Card, SeriesOrderSetting } from "metabase-types/api";
+
+export const getDefaultStackingValue = (
+  settings: ComputedVisualizationSettings,
+  card: Card,
+) => {
+  // legacy setting and default for D-M-M+ charts
+  if (settings["stackable.stacked"]) {
+    return settings["stackable.stacked"];
+  }
+
+  const shouldStack =
+    card.display === "area" &&
+    ((settings["graph.metrics"] ?? []).length > 1 ||
+      (settings["graph.dimensions"] ?? []).length > 1);
+
+  return shouldStack ? "stacked" : null;
+};
+
+export const getSeriesOrderVisibilitySettings = (
+  settings: ComputedVisualizationSettings,
+  seriesKeys: string[],
+) => {
+  const seriesSettings = settings["series_settings"];
```

Comment:
```suggestion
  const seriesSettings = settings[SERIES_SETTING_KEY];
```

We should use the constants defined in `series.ts`